lang/cs/Org.Apache.REEF.Client/YARN/IJobResourceUploader.cs

```diff
@@ -26,7 +26,8 @@ public interface IJobResourceUploader
         /// Upload archived local driver folder to DFS destination path.
         /// </summary>
         /// <param name="driverLocalFolderPath">Local folder where REEF application resources are staged</param>
+        /// <param name="jobSubmissionDirectory">Remote directory path where we will upload resources</param>
         /// <returns>Path, modification time and size of uploaded file as JobResource</returns>
-        JobResource UploadJobResource(string driverLocalFolderPath);
+        JobResource UploadJobResource(string driverLocalFolderPath, string jobSubmissionDirectory);
```

Comment:
One instance of uploader should be able to upload in different locations, if need by. Earlier we thought we would generate `jobSubmissionDIrectory` in `IJobResourceUploader` which was a bad idea in hindsight, creating hidden coupling between `IJobSubmissionDirectoryProvider` and `IJobResourceUploader`.
I would strongly prefer not to go the `Obsolete` route here as it is not possible for anyone to take dependency on this API right now. It is just added work for no good reason. Thoughts?